flask_praetorian/base.py

```diff
@@ -412,6 +448,9 @@ def encode_jwt_token(
             "Attaching custom claims: {}".format(custom_claims),
         )
         payload_parts.update(custom_claims)
+
+        if self.encode_jwt_token_hook:
+            self.encode_jwt_token_hook(copy.deepcopy(payload_parts))
```

Comment:
Instead of passing the dict of payload_parts to the hook, it would be better (for readability and immutability) to pass the unrolled dict:

```
if self.encode_jwt_token_hook is not None:
    self.encode_jwt_token_hook(**payload_parts)
```

This would remove the need for copy/deepcopy. The only part that risks being modified in the hook is the custom_claims. Since the client code is responsible for their own custom_claims, this doesn't concern me too much.

Reply:
I like it! Yes, I was specifically thinking about custom_claims (as well) given that nested json is supported but completely agree on the client responsibility part.